File: src/types/segment.ts
export interface Segment {
  id: string;
  task_id: string;
  start_time: number;
  end_time: number;
  text: string;
  speaker_id?: string;
  speaker_name?: string;
  confidence?: number;
  is_crosstalk: boolean;
  is_checked: boolean;
  status: 'pending' | 'edited' | 'verified' | 'rejected';
  comments?: Comment[];
  created_at: string;
  updated_at: string;
}

export interface Comment {
  id: string;
  segment_id: string;
  author_id: string;
  author_name: string;
  text: string;
  status: 'open' | 'resolved';
  created_at: string;
}

File: backend/internal/models/segment.go
package models

type Segment struct {
	ID        int64   `json:"id"`
	TaskID    int64   `json:"task_id"`
	StartTime float64 `json:"start_time"`
	EndTime   float64 `json:"end_time"`
	Text      string  `json:"text"`
	IsChecked bool    `json:"is_checked"`
}